src/cdk/datetime/datepicker-input.ts

```diff
@@ -0,0 +1,345 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {
+  AfterContentInit,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Inject,
+  Input,
+  OnDestroy,
+  Optional,
+  Output,
+} from '@angular/core';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+  ValidatorFn,
+  Validators
+} from '@angular/forms';
+import {DateAdapter} from './date-adapter';
+import {Subscription} from 'rxjs';
+import {CdkDatepicker} from './datepicker';
+import {CDK_DATE_FORMATS, CdkDateFormats} from './date-formats';
+
+
+/**
+ * Provider that allows the datepicker to register as a ControlValueAccessor.
+ */
+export const DATEPICKER_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * Provider that allows the datepicker to register as a ControlValueAccessor.
+ */
+export const DATEPICKER_VALIDATORS: any = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * An event used for datepicker input and change events. For consistency, we always use
+ * DatepickerInputEvent instead.
+ */
+export class DatepickerInputEvent<D> {
+  /** The new value for the target datepicker input. */
+  value: D | null;
+
+  constructor(
+      /** Reference to the datepicker input component that emitted the event. */
+      public target: CdkDatepickerInput<D>,
+      /** Reference to the native input element associated with the datepicker input. */
+      public targetElement: HTMLElement) {
+    this.value = this.target.value;
+  }
+}
+
+
+/** Directive used to connect an input to a CdkDatepicker. */
+@Directive({
+  selector: 'input[cdkDatepicker]',
+  providers: [
+    DATEPICKER_VALUE_ACCESSOR,
+    DATEPICKER_VALIDATORS,
+  ],
+  host: {
+    '[attr.aria-owns]': '(_datepicker.id) || null',
```

Comment:
I think this should have `aria-haspopup` as well

Reply:
I was thinking we'd add this in the material one, since you could make one with a calendar that's always shown, not a popup